src/prpy/serialization.py

```diff
@@ -99,43 +101,60 @@ def serialize_environment_file(env, path, writer=None):
 
     return data
 
-def serialize_kinbody(body):
-    all_joints = []
-    all_joints.extend(body.GetJoints())
-    all_joints.extend(body.GetPassiveJoints())
+def serialize_kinbody(body, uri_only=False):
 
+    uri = body.GetXMLFilename()
+    
+    if not uri:
+        uri_only = False
```

Comment:
Print a warning in this case. It's odd that you can pass `uri_only`, but then have the implementation silently fall back on full serialization. Or add another option like `try_uri` that makes it clear that it's best-effort only.